idaes/power_generation/carbon_capture/mea_solvent_system/property_package/liquid_prop.py

```diff
@@ -0,0 +1,1282 @@
+                Fb = blk._params.surf_tens_F_param[2]
+                Fc = blk._params.surf_tens_F_param[3]
+                Fd = blk._params.surf_tens_F_param[4]
+                Fe = blk._params.surf_tens_F_param[5]
+                Ff = blk._params.surf_tens_F_param[6]
+                Fg = blk._params.surf_tens_F_param[7]
+                Fh = blk._params.surf_tens_F_param[8]
+                Fi = blk._params.surf_tens_F_param[9]
+                Fj = blk._params.surf_tens_F_param[10]
+                return blk.surf_tens_comp['H2O'] + \
+                    (blk.surf_tens_comp['CO2'] - blk.surf_tens_comp['H2O'])*\
+                    blk.mole_frac['CO2']*(Fa+Fb*alpha+Fc*alpha**2+Fd*r+Fe*r**2)\
+                    + (blk.surf_tens_comp['MEA'] - blk.surf_tens_comp['H2O'])*\
+                   (Ff + Fg*alpha + Fh*alpha**2 + Fi*r + Fj*r**2)*\
+                   blk.mole_frac['MEA']
+
+        self.surf_tens = Expression(rule = rule_surf_tens,
+                                    doc="surface tension [N/m]")
+
+    def _pressure_sat(self):
+        '''
+        saturated vapor pressure of solvents
+        '''
+        def rule_pressure_sat_comp(blk,j):
+                T = blk.temperature
+                return (exp(blk._params.pressure_sat_param[j,1]
+                          + blk._params.pressure_sat_param[j,2]/T +
+                            blk._params.pressure_sat_param[j,3]*log(T) +
+                            blk._params.pressure_sat_param[j,4]*T*T))
+
+        self.pressure_sat = Expression(self._params.component_list_solvent,
+                                       rule = rule_pressure_sat_comp,
+                                    doc="Vapor pressure [Pa]")
+
+    def _diffus(self):
+        """
+        Diffusivity of diffusing components
+        """
+        def rule_diffus(blk, i):
+            if i == 'CO2':
+              C_MEA = blk.conc_comp['MEA']*1e-3
+              return (blk._params.diffus_param[i,1] +
+                    blk._params.diffus_param[i,2]*C_MEA +
+                    blk._params.diffus_param[i,3]*C_MEA**2)*exp(
+                    (blk._params.diffus_param[i,4] +
+                     blk._params.diffus_param[i,5])/blk.temperature)
+            elif i == 'MEA':
+               C_MEA = blk.conc_comp['MEA']
+               return exp(blk._params.diffus_param[i,1] +
+                          blk._params.diffus_param[i,2]/blk.temperature +
+                          blk._params.diffus_param[i,3]*C_MEA)
+            elif i == "MEACOO-" or i == 'MEA+':
+               return exp(blk._params.diffus_param[i,1] +
+                          blk._params.diffus_param[i,2]/blk.temperature +
+                          blk._params.diffus_param[i,3]*log(blk.visc_d))
+
+        self.diffus = Expression(self._params.component_list_diffus,
+                                 rule=rule_diffus,
+                             doc='diffusivity of component i in liquid [m^2/s]')
+
+    def _k2_rxn(self):
```

Comment:
I will need to look at how the unit model is set up, but my first feeling is that rate expressions belong in a Reaction Block, not in a State Block.

Reply:
This overall reaction rate is used in the enhancement factor model for calculating the Hatta number only.